src/vss_tools/vspec/vssexporters/vss2apigear.py

```diff
@@ -0,0 +1,522 @@
+            if prop is not None and interface is not None:
+                interface.properties[node.name] = prop
+
+
+def export_data_type_node(node: VSSNode | None, module: ApiGearModule, structure: ApiGearStructure | None = None):
+    """This method is used to traverse through the root VSS Data Types node to build
+       -> ApiGear equivalent string buffer and to serialize it accordingly into a module
+    """
+    if node is None:
+        return
+
+    log.debug(f"Node [{node.type}] {node.qualified_name()}")
+
+    if node.type == VSSType.BRANCH:
+        for child in node.children:
+            export_data_type_node(child, module)
+    elif node.type == VSSType.STRUCT:
+        struct_name = node_name(node)
+        struct = ApiGearStructure(struct_name)
+        log.debug(f"Creating a new structure {struct_name}")
+
+        for child in node.children:
+            export_data_type_node(child, module, struct)
+
+        module.structures.append(struct)
+    else:
+        if node.has_datatype() and node.datatype is not None:
+            datatype = node.datatype.value
+        elif hasattr(node, "data_type_str"):
+            datatype = node.data_type_str
+        else:
+            log.warning(f"Node {node.qualified_name()} does not have a datatype")
+            return
+        prop = generate_property(node, datatype)
+        if prop is not None and structure is not None:
+            structure.properties[node.name] = prop
+
+
+def generate_solution(directory: Path, module_filename: str, module_name: str,
+                      layers: typing.Dict[SolutionLayers, Path]):
+    log.info("Generating solution file")
+
+    path = os.path.join(directory, SOLUTION_FILENAME.format(module_name))
+    file = open(path, "w")
+
+    file.write(SOLUTION_TEMPLATE_START.format(module_name))
+
+    if (SolutionLayers.UNREAL in layers):
+        file.write(SOLUTION_TEMPLATE_LAYER_UNREAL
+                   .format(module_name=module_filename, output_path=layers[SolutionLayers.UNREAL]))
+    if (SolutionLayers.CPP in layers):
+        file.write(SOLUTION_TEMPLATE_LAYER_CPP
+                   .format(module_name=module_filename, output_path=layers[SolutionLayers.CPP]))
+    if (SolutionLayers.QT5 in layers):
+        file.write(SOLUTION_TEMPLATE_LAYER_QT5
+                   .format(module_name=module_filename, output_path=layers[SolutionLayers.QT5]))
+    if (SolutionLayers.QT6 in layers):
+        file.write(SOLUTION_TEMPLATE_LAYER_QT6
+                   .format(module_name=module_filename, output_path=layers[SolutionLayers.QT6]))
+
+    log.info("Solution file generated at location : " + path)
+
+
+def generate_module(file, module: ApiGearModule, module_name: str):
+    file.write(MODULE_TEMPLATE_START.format(module_name))
+    file.write("interfaces:\n")
+
+    for interface in module.interfaces:
+        if len(interface.properties) == 0:
+            continue
+
+        file.write(f"  - name: {interface.name}\n")
```

Comment:
are we trying to write yaml here?
Why are we not building a correct dict and using yaml dump?
Generating yaml with raw strings looks a bit dangerous

Reply:
Right, I'll rework it using the yaml module.

Reply:
Reworked it using the yaml module.